core/storage/btree.rs

```diff
@@ -145,57 +146,103 @@ impl BTreeCursor {
                     let record = crate::storage::sqlite3_ondisk::read_record(_payload)?;
                     return Ok(CursorResult::Ok((Some(*_rowid), Some(record))));
                 }
-                BTreeCell::IndexInteriorCell(_) => {
-                    unimplemented!();
+                BTreeCell::IndexInteriorCell(IndexInteriorCell {
+                    payload,
+                    left_child_page,
+                    ..
+                }) => {
+                    mem_page.advance();
+                    let mem_page =
+                        MemPage::new(Some(mem_page.clone()), *left_child_page as usize, 0);
+                    self.page.replace(Some(Rc::new(mem_page)));
+                    continue;
                 }
-                BTreeCell::IndexLeafCell(_) => {
-                    unimplemented!();
+                BTreeCell::IndexLeafCell(IndexLeafCell { payload, .. }) => {
+                    mem_page.advance();
+                    let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
+                    let rowid = match record.values.last() {
+                        Some(OwnedValue::Integer(rowid)) => *rowid as u64,
```

Comment:
Aren't index keys also blobs/strings/whatever?

Reply:
We are getting the rowid here and AFAIK it's always the last value in the payload record